src/core/server/saved_objects/migrationsv2/integration_tests/multiple_kibana_nodes.test.ts

```diff
@@ -0,0 +1,219 @@
+        batchSize: 100, // fixture contains 5000 docs
+      },
+      logging: {
+        appenders: {
+          file: {
+            type: 'file',
+            fileName: logFileName,
+            layout: {
+              type: 'pattern',
+            },
+          },
+        },
+        loggers: [
+          {
+            name: 'root',
+            appenders: ['file'],
+          },
+          {
+            name: 'savedobjects-service',
+            appenders: ['file'],
+            level: 'debug',
+          },
+        ],
+      },
+    },
+    {
+      oss: true,
+    }
+  );
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let rootA: Root;
+  let rootB: Root;
+  let rootC: Root;
+
+  const migratedIndex = `.kibana_${pkg.version}_001`;
+  const fooType: SavedObjectsType = {
+    name: 'foo',
+    hidden: false,
+    mappings: { properties: { status: { type: 'text' } } },
+    namespaceType: 'agnostic',
+    migrations: {
+      '7.14.0': (doc) => {
+        if (doc.attributes?.status) {
+          doc.attributes.status = doc.attributes.status.replace('not_migrated', 'migrated');
+        }
+        return doc;
+      },
+    },
+  };
+
+  beforeAll(async () => {
+    await removeLogFiles();
+  });
+
+  afterAll(async () => {
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  beforeEach(async () => {
+    rootA = createRoot({
+      logFileName: Path.join(__dirname, `${LOG_FILE_PREFIX}_A.log`),
+    });
+    rootB = createRoot({
+      logFileName: Path.join(__dirname, `${LOG_FILE_PREFIX}_B.log`),
+    });
+    rootC = createRoot({
+      logFileName: Path.join(__dirname, `${LOG_FILE_PREFIX}_C.log`),
+    });
+
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'trial',
+          // original SO:
+          // [
+          //   { id: 'foo:1', type: 'foo', foo: { status: 'not_migrated_1' } },
+          //   { id: 'foo:2', type: 'foo', foo: { status: 'not_migrated_2' } },
+          //   { id: 'foo:3', type: 'foo', foo: { status: 'not_migrated_3' } },
+          // ];
+          dataArchive: Path.join(__dirname, 'archives', '7.13.0_5k_so.zip'),
+        },
+      },
+    });
+    esServer = await startES();
+  });
+
+  afterEach(async () => {
+    await Promise.all([rootA.shutdown(), rootB.shutdown(), rootC.shutdown()]);
+
+    if (esServer) {
+      await esServer.stop();
+    }
+  });
+
+  it('migrates saved objects normally when multiple Kibana instances are running', async () => {
+    const setupContracts = await Promise.all([rootA.setup(), rootB.setup(), rootC.setup()]);
+
+    setupContracts.forEach((setup) => setup.savedObjects.registerType(fooType));
+
+    const startContracts = await Promise.all([rootA.start(), rootB.start(), rootC.start()]);
+
+    const esClient = startContracts[0].elasticsearch.client.asInternalUser;
```

Comment:
```js
const esClient = esServer.es.getClient();
```
to make sure the test depends on test utils only.

Reply:
Didn't know about `getClient`, thanks!